### PR TITLE
Amanda billing

### PR DESCRIPTION
**Pull request by phillipsdev**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change
Billing page (sans styling) integrated with Stripe updates user model to premium status upon successful completion of payment.

- dependencies server: stripe
- dependencies client: react-stripe-elements, ajv
("ajv" was a peer dependency that was missing, and had to be installed)

The client Stripe apiKey is my personal apiKey and as it states in the comment, it is a publishable key, however it will need to be replaced if you are wanting to test out on your side.

Server config - STRIPE_API_KEY_TEST
I named it test, so that I would remember that it was a test key, in production we will change the variable.

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Charged Stripe test cards on multiple user accounts to verify that Stripe was working accurately and premiumUser value was updated

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: server/api/controllers/UserControllers.js [hunk 1]
```diff
@@ -220,6 +225,38 @@ const changeEmail = (req, res) => {
   );
 };
 
+// using async and await according to Stripe docs,
+// I had a hard time getting it to play nice otherwise
+const processPayment = async (req, res) => {
```
Comment (rusticlion) [inline server/api/controllers/UserControllers.js:230]: Honestly, I wouldn't be opposed to refactoring to `async/await` app-wide - I personally prefer it to .then. WE can discuss in standup

File: server/api/models/User.js [hunk 2]
```diff
@@ -64,7 +64,11 @@ const UserSchema = Schema({
         ref: "Workout"
       }
     }
-  ]
+  ],
+  premiumUser: {
```
Comment (rusticlion) [inline server/api/models/User.js:68]: If we want to expand this to a recurring charge for the service, all we would have to do is expand this field to keep track of their subscription status.

**Review by rusticlion — APPROVED** [review]
Looks good style-wise and works on my machine!

**Review by emukupa — APPROVED** [review]
looks great! Keep up the good work 👍

**Pull request merged**